src/VisualStudio/Core/Def/Implementation/ProjectSystem/DocumentProvider.TextBufferDataEventsSink.cs

```diff
@@ -40,7 +56,7 @@ public int OnLoadCompleted(int fReload)
             {
                 _sink.Unadvise();
 
-                _documentProvider.DocumentLoadCompleted(_textBuffer, _documentKey);
+                _documentProvider.DocumentLoadCompleted(_textBuffer, _documentKey, _moniker);
```

Comment:
Rather than coupling these types further, I wonder if we're better off this just making this type take a lambda it invokes when the load is complete.